### ISSUE TITLE
provider must be in args, not kwargs

### ISSUE THREAD

**Issue by root-11** [ponyorm/pony#298]
Following the api reference AND the tutorials, the usage:

`self.db.bind(provider, filename=filename, create_db=True)
`
is okay, whilst the usage:

`self.db.bind(provider=provider, filename=filename, create_db=True)
`
throws the exception:

```
Connected to pydev debugger (build 172.3968.37)
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2017.2.3\helpers\pydev\pydevd.py", line 1599, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "C:\Program Files\JetBrains\PyCharm 2017.2.3\helpers\pydev\pydevd.py", line 1026, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2017.2.3\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:/dce_v1/tools/dce_db/dce_db/tests.py", line 28, in <module>
    doall()
  File "D:/dce_v1/tools/dce_db/dce_db/tests.py", line 24, in doall
    v()
  File "D:/dce_v1/tools/dce_db/dce_db/tests.py", line 6, in test01
    s = Database(as_test=True)
  File "D:\dce_v1\tools\dce_db\dce_db\model.py", line 197, in __init__
    self.db.bind(provider=provider, filename=filename, create_db=True)
  File "<string>", line 2, in bind
  File "C:\Anaconda3\envs\dce\lib\site-packages\pony\utils\utils.py", line 58, in cut_traceback
    return func(*args, **kwargs)
  File "C:\Anaconda3\envs\dce\lib\site-packages\pony\orm\core.py", line 540, in bind
    self._bind(*args, **kwargs)
  File "C:\Anaconda3\envs\dce\lib\site-packages\pony\orm\core.py", line 546, in _bind
    throw(TypeError, 'Database provider should be specified as a first positional argument')
  File "C:\Anaconda3\envs\dce\lib\site-packages\pony\utils\utils.py", line 96, in throw
    raise exc
TypeError: Database provider should be specified as a first positional argument
```

This is a silly error. Either use args or kwargs consistently.

Personally I would suggest to patch the code in the "Database.bind" method:

```
@cut_traceback
    def bind(self, *args, **kwargs):
        self._bind(*args, **kwargs)
    def _bind(self, *args, **kwargs):
        # argument 'self' cannot be named 'database', because 'database' can be in kwargs
        if self.provider is not None:
            throw(TypeError, 'Database object was already bound to %s provider' % self.provider.dialect)
        if not args:
            throw(TypeError, 'Database provider should be specified as a first positional argument')
        provider, args = args[0], args[1:]
        if isinstance(provider, type) and issubclass(provider, DBAPIProvider):
            provider_cls = provider
        else:
            if not isinstance(provider, basestring): throw(TypeError)
            if provider == 'pygresql': throw(TypeError,
                'Pony no longer supports PyGreSQL module. Please use psycopg2 instead.')
            provider_module = import_module('pony.orm.dbproviders.' + provider)
            provider_cls = provider_module.provider_cls
        self.provider = provider = provider_cls(*args, **kwargs)

```

With:

```
@cut_traceback
    def bind(self, *args, **kwargs):
        self._bind(*args, **kwargs)
    def _bind(self, *args, **kwargs):
        # argument 'self' cannot be named 'database', because 'database' can be in kwargs
        if self.provider is not None:
            throw(TypeError, 'Database object was already bound to %s provider' % self.provider.dialect)
        if 'provider' not in kwargs:
            if not args:
                throw(TypeError, 'Database provider should be specified as a first positional argument')
                try:
                    provider, args = args[0], args[1:]
                    if isinstance(provider, type) and issubclass(provider, DBAPIProvider):
                        provider_cls = provider
                 except Exception:
                     raise UnCaughtException("please post this to github...")
            else:
                if not isinstance(provider, basestring): throw(TypeError)
                if provider == 'pygresql': throw(TypeError,
                    'Pony no longer supports PyGreSQL module. Please use psycopg2 instead.')
                provider_module = import_module('pony.orm.dbproviders.' + provider)
                provider_cls = provider_module.provider_cls
        self.provider = provider = provider_cls(*args, **kwargs)
```

**Comment by kozlovsky** [comment]
Hi, it was already fixed in 0.7.2 release, see commit f86bd3ae

**Comment by root-11** [comment]
confirmed. 
Issue can be closed.
